<?xml version="1.0"?>
<!-- Provided by sleutene for convenience -->
<launch>
  <!-- Launch simulator. -->
  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <include file="$(find sjtu_drone)/launch/simple.launch">

    <arg name="world_name" value="$(find sjtu_drone)/worlds/contest.world"/>
  </include>
  
  <!-- Emulate controller parameters-->
  <group ns="ardrone_driver">
    <param name="euler_angle_max" value="0.21" />
    <param name="control_vz_max" value="700" />
    <param name="control_yaw" value="1.75" />
  </group>

  <!-- Launch visualisation. -->
  <!--node pkg="rviz" type="rviz" name="rviz" args="-d $(find ardrone_practicals)/config/rviz_vi_ekf.rviz"/-->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find ardrone_practicals)/config/rviz_control.rviz"/>

  <!-- Set Log Level, possible levels: debug, info, warn, error, fatal -->
  <node pkg="rosservice" type="rosservice" name="set_node_log_level" 
      args="call /arp_node/set_logger_level '{logger: 'ros.ardrone_practicals', level: 'info'}'" />
  <node pkg="rosservice" type="rosservice" name="set_log_level" 
      args="call /arp_node/set_logger_level '{logger: 'ros.ardrone_practicals.custom', level: 'info'}'" />

  <!-- Launch estimator / control arp_node. -->
  <node name="arp_node" pkg="ardrone_practicals" type="arp_node"
      output="screen" args="log_level:=debug">
    <remap from="ardrone/front/image_raw" to="drone/front_camera/image_raw"/>
    <remap from="ardrone/imu" to="drone/imu"/> 
    <remap from="ardrone/takeoff" to="drone/takeoff"/> 
    <remap from="ardrone/reset" to="drone/reset"/>
    <remap from="ardrone/land" to="drone/land"/> 
    <remap from="ardrone/navdata" to="drone/navdata"/> 

    <!-- Global Variables -->
    <param name="enableFusion" type="bool" value="true" />
    <param name="camModel" type="bool" value="false" />
    <param name="displayKeypoints" type="bool" value="true" />
    <param name="displayAllKeypoints" type="bool" value="false" />
    <param name="numKeypoints" value="200" />
    <param name="poseLostThreshold" value="100" />
    <param name="poseSwitchThreshold" value="5" />
    <param name="poseLostTimeThreshold" value="2" />
    <param name="mapFocalLength" value="185.6909" />
    <param name="Brisk_uniformityRadius" value="100" />
    <param name="Brisk_absoluteThreshold" value="30" />
    <param name="ImageWidth" value="1920" />
    <param name="ImageHeight" value="960" />
    <param name="fontScaling" value="1.5" />
    <param name="skipThresInit" value="3" /> <!-- default 1 -->
    <param name="skipThresLimit" value="3" /> <!-- default 4 -->

    <!-- WaypointB -->
    <param name="waypointX" value="100" />
    <param name="waypointY" value="100" />
    <param name="waypointZ" value="100" />
    <param name="waypointYaw" value="0" />
    <param name="waypointTolerance" value="10" />  
  
    <!-- Camera parameters -->
    <param name="fu" value="185.6909" />
    <param name="fv" value="185.6909" />
    <param name="cu" value="319.5" />
    <param name="cv" value="179." />
    <param name="k1" value="0" />
    <param name="k2" value="0" />
    <param name="p1" value="0" />
    <param name="p2" value="0" />
    <rosparam param="T_SC">[0.0, 0.0, 1.0, 0.2, -1.0, 0.0, 0.0, 0.0, 0.0, -1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0]</rosparam>
    <!-- Make sure to match the map to the world loaded above! -->
    <param name="map" value="okvis2-slam-final_map_dronelab.csv" />
    <param name="occupancymap" value="occupancy-map_skokloster.dat" />

  </node>

</launch>
